### PR TITLE
Feature/total time screen

### PR DESCRIPTION
**Pull request by 2Fick**
# **Description**
- **What has been changed?**  
  A new screen (`TimePickerScreen`) has been added to the recipe creation flow. It allows users to optionally input the total preparation time for their recipe using hour and minute selectors. The selected time is stored in minutes in the `CreateRecipeViewModel` and integrates seamlessly into the existing navigation flow.
image

- **Why was this change made?**  
  This feature was added to enhance the recipe creation process, enabling users to provide an optional preparation time for better recipe context.

---

# **Code Snippets**

### **`WheelTimePicker`**
This composable uses two `NumberPickerComposable` components to let users select hours and minutes independently. A `Text` component displays a colon (`:`) separator between the hour and minute pickers. The layout is responsive and adapts to various screen widths.

**Example Usage:**
```kotlin
WheelTimePicker(
    selectedHour = 12,
    selectedMinute = 30,
    onTimeSelected = { hour, minute ->
        println("Selected time: $hour:$minute")
    }
)
```

### **`NumberPickerComposable`**
This composable uses `AndroidView` to integrate the native Android `NumberPicker` widget into a Compose layout. It exposes the current value as a semantic property for testing and accessibility. The value is dynamically updated when the `value` parameter changes.

**Example Usage:**
```kotlin
NumberPickerComposable(
    value = 5,
    range = 0..10,
    onValueChange = { newValue ->
        println("New value: $newValue")
    }
)
```

---

# **Problems Encountered**
1. **Initial Attempt Using a Third-Party Library:**  
   Initially tried implementing the picker using [WheelPickerCompose](https://github.com/commandiron/WheelPickerCompose/tree/master). However, this library is under the Apache-2.0 license, which required us to abandon the implementation to avoid licensing conflicts.  

2. **UI Testing with Native `NumberPicker`:**  
   The native Android `NumberPicker` does not expose its values directly to Compose's testing APIs. To resolve this, semantic properties were added to the `NumberPickerComposable` to expose the current value for testing purposes.  

---

# **Checklist**
- [x] Tests added.  
  Unit and E2E tests have been added for `TimePickerScreen` and its integration with the recipe creation flow.

- [x] Documentation updated.  
  Inline comments and descriptions have been added for `TimePickerScreen`, `WheelTimePicker`, and `NumberPickerComposable`.

- [x] All CI checks passed.  
  Verified that all tests, linting, and build processes pass successfully with coverage > 80% on `SonarCloud`.

- [x] Linked issue/feature:  
  #231

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/createRecipe/TimePickerScreen.kt [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+          Spacer(modifier = Modifier.weight(6f))
+        }
+
+    // Next Step Button
+    PlateSwipeButton(
+        text = stringResource(R.string.next_step),
+        modifier = Modifier.align(Alignment.BottomCenter).testTag(NEXT_BUTTON),
+        onClick = {
+          val totalTimeInMinutes = (hours.intValue * 60) + minutes.intValue
```
Comment (akshire) [inline app/src/main/java/com/android/sample/ui/createRecipe/TimePickerScreen.kt:137]: Can you put this value in the `C` file

File: app/src/main/java/com/android/sample/ui/createRecipe/TimePickerScreen.kt [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+@Composable
+fun TimePickerContent(
+    modifier: Modifier = Modifier,
+    currentStep: Int,
+    navigationActions: NavigationActions,
+    createRecipeViewModel: CreateRecipeViewModel
+) {
+  val totalMinutes = createRecipeViewModel.getRecipeTime()?.toIntOrNull() ?: 0
+  val initialHours = totalMinutes / 60
+  val initialMinutes = totalMinutes % 60
```
Comment (akshire) [inline app/src/main/java/com/android/sample/ui/createRecipe/TimePickerScreen.kt:80]: these constants should be in the `C` file

**Review by akshire — CHANGES_REQUESTED** [review]
Great job on this PR! 🚀 The feature is incredible, and the Recipe Creation screen truly enhances the user experience. 🙌

While testing the UI, I came across a few minor issues that could be addressed to make this even better:

1. Scrolling Issue: When attempting to scroll, the time remains static and doesn't scroll properly. Here's a screen capture for reference:

https://github.com/user-attachments/assets/8dbe3674-e0fd-4bc1-8856-6431b416ec15

2. Time Reset Issue: If I input a time, navigate back, and return to the time screen, the previously entered time gets reset. This disrupts the flow for users. Here's a screen capture of the behavior:

https://github.com/user-attachments/assets/bba1bc56-5d14-4fc6-9413-1c8e6bd71824

3. Progress Bar Consistency: The Progress Bar design on this screen doesn’t match the one on the previous screen. It’s a small detail but maintaining consistency in UI elements helps with a cohesive user experience.

Overall, fantastic work! This feature is shaping up to be a valuable addition. With these tweaks, it will be even more polished. Keep up the great work! 🎉

File: app/src/main/java/com/android/sample/ui/createRecipe/TimePickerScreen.kt [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+          Text(
+              text = stringResource(R.string.hours),
+              style = MaterialTheme.typography.bodySmall,
+              color = MaterialTheme.colorScheme.onPrimary,
+              modifier = Modifier.testTag(HOURS_LABEL))
+
+          // Hour Picker
+          NumberPickerComposable(
+              value = selectedHour,
+              range = 0..23,
```
Comment (akshire) [inline app/src/main/java/com/android/sample/ui/createRecipe/TimePickerScreen.kt:185]: This one should also be in the C file

File: app/src/main/java/com/android/sample/ui/createRecipe/TimePickerScreen.kt [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+          Text(
+              text = stringResource(R.string.minutes),
+              style = MaterialTheme.typography.bodySmall,
+              color = MaterialTheme.colorScheme.onPrimary,
+              modifier = Modifier.testTag(MINUTES_LABEL))
+
+          // Minute Picker
+          NumberPickerComposable(
+              value = selectedMinute,
+              range = 0..59,
```
Comment (akshire) [inline app/src/main/java/com/android/sample/ui/createRecipe/TimePickerScreen.kt:209]: This one should also be in the C file

**Review by akshire — CHANGES_REQUESTED** [review]
I must say that the code is really clean but I forgot to point you these two constants that are not in the C file.

**Comment by 2Fick** [comment]
> Great job on this PR! 🚀 The feature is incredible, and the Recipe Creation screen truly enhances the user experience. 🙌
> 
> While testing the UI, I came across a few minor issues that could be addressed to make this even better:
> 
> 1. Scrolling Issue: When attempting to scroll, the time remains static and doesn't scroll properly. Here's a screen capture for reference:
> 
>  Screen_Recording_2024-11-28_180520.mp4 
> 2. Time Reset Issue: If I input a time, navigate back, and return to the time screen, the previously entered time gets reset. This disrupts the flow for users. Here's a screen capture of the behavior:
> 
>  Screen_Recording_2024-11-28_180436.mp4 
> 3. Progress Bar Consistency: The Progress Bar design on this screen doesn’t match the one on the previous screen. It’s a small detail but maintaining consistency in UI elements helps with a cohesive user experience.
> 
> Overall, fantastic work! This feature is shaping up to be a valuable addition. With these tweaks, it will be even more polished. Keep up the great work! 🎉

For the first point, i didnt catch the problem. But 2nd and 3rd point have been resolved.

**Comment by akshire** [comment]
Regarding the first issue, when watching the video, the scrolling seems quite choppy; it stops after scrolling just three digits. The idea would be to make it smoother and have it stop after more than three digits to bring a sense of fluidity to the scrolling.

**Review by akshire — APPROVED** [review]
Regarding the first issue, when watching the video, the scrolling seems quite choppy; it stops after scrolling just three digits. The idea would be to make it smoother and have it stop after more than three digits to bring a sense of fluidity to the scrolling.

But this can be solved in a future PR, great job! Ready to be merged 🚀

**Pull request merged**